platform/fdo/cog-platform-fdo.c

```diff
@@ -2092,18 +2092,68 @@ clear_wayland (void)
     wl_display_disconnect (wl_data.display);
 }
 
+#define SHELL_PROTOCOLS(m) \
+    m(xdg_wm_base) \
+    m(wl_shell) \
+    m(zwp_fullscreen_shell_v1)
+
+#define DECLARE_PROTOCOL_ENTRY(proto) struct proto *proto;
+struct check_supported_protocols { SHELL_PROTOCOLS(DECLARE_PROTOCOL_ENTRY) };
+#undef DECLARE_PROTOCOL_ENTRY
+
+static void
+on_registry_global_is_supported_check(void *data, struct wl_registry *registry,
+                                      uint32_t name, const char *interface,
+                                      uint32_t version)
+{
+    struct check_supported_protocols *protocols = data;
+
+#define TRY_BIND_PROTOCOL_ENTRY(proto) \
+    if (strcmp(interface, proto ## _interface.name) == 0) { \
+        protocols->proto = wl_registry_bind(registry, name, &proto ## _interface, version); \
```

Comment:
Binding shouldn't be needed. But whatever is bound has to be cleaned up.

Reply:
You are right, it's enough to know that the protocol is supported by the compositor. We can have a boolean flag and toggle it when the interface name is matched.

Reply:
...and done!